### ISSUE TITLE
Shim 15.6 for Trusted Disk

### ISSUE THREAD

**Issue by ifankner** [rhboot/shim-review#257]
Confirm the following are included in your repo, checking each box:

 - [x] completed README.md file with the necessary information  
 https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/README.md
 - [x] shim.efi to be signed  
 https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/shimx64.efi  
 - [x] public portion of your certificate(s) embedded in shim (the file passed to VENDOR_CERT_FILE)  
 https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/rs-uefi-cs.der
 - [x] binaries, for which hashes are added to vendor_db ( if you use vendor_db and have hashes allow-listed )  
 N/A
 - [x] any extra patches to shim via your own git tree or as files  
https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/patches/msabi.patch  
https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/patches/bypass_bootoptions.patch 
 - [x] any extra patches to grub via your own git tree or as files  
 N/A 
 - [x] build logs  
https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/build.log
 - [x] a Dockerfile to reproduce the build of the provided shim EFI binaries  
https://github.com/Rohde-Schwarz/shim-review/blob/rohdeschwarz-shim-x64-20220630/Dockerfile

-------------------------------------------------------------------------------
### What is the link to your tag in a repo cloned from rhboot/shim-review?
-------------------------------------------------------------------------------
`https://github.com/Rohde-Schwarz/shim-review/tree/rohdeschwarz-shim-x64-20220630`

-------------------------------------------------------------------------------
### What is the SHA256 hash of your final SHIM binary?
-------------------------------------------------------------------------------
`2F0FAA6451D33E05FB12CD43EBA618563C40EA5CF531ABD5AB7CC74F53132FBB`

**Comment by ifankner** [comment]
First of all, thanks for the review.

And you are right, we moved forward to Shim 15.6. "No changes were made" referred to the fact, that we didn't add extra patches to the Shim code since the last submission of our signing request :-)

As I mentioned it in the README file, the first patch "bypass_bootoptions.patch" bypasses the bootoption parsing and second stage loader name assembly. We noticed an "invalid parameter" error on some Dell notebooks (e.g. 7470, 7470, 7480 Touch) with the 15.4 shim code. The bootoption parsing code failed on these systems and produced a non-bootable name for the second stage loader. And so we just bypass this problem with this patch, because we are using a hardcoded path and filename for the second stage loader and don't need this feature.

We need the second patch for our pre-boot authentication code, which is built with the Microsoft Visual Compiler. To access the SHIM LOCK verification protocol, we need to use the same ABI on both sides. Unfortunately the Microsoft Visual Compiler is not able to use the linux System V ABI, and so we had to switch to MS ABI.

Yes, tdisk (Trusted Disk) is our pre-boot authentication EFI software. With a smartcard and a corresponding PIN, a key is derived to decrypt the whole full-disk-encrypted harddisk. The only external software which can be loaded is the windows bootloader. The integrity of each file in our boot chain is verified by checking the validity of the digital signature when using the LoadImage/StartImage functions. And so it is not possible to load any unsigned or revoked code.
